homeassistant/components/home_connect/switch.py

```diff
@@ -182,42 +181,45 @@ async def async_turn_off(self, **kwargs: Any) -> None:
                 },
             ) from err
 
-        self._attr_available = True
-        self.async_entity_update()
-
-    async def async_update(self) -> None:
+    def update_native_value(self) -> None:
         """Update the switch's status."""
-
-        self._attr_is_on = self.device.appliance.status.get(
-            self.entity_description.key, {}
-        ).get(ATTR_VALUE)
-        self._attr_available = True
-        _LOGGER.debug(
-            "Updated %s, new state: %s",
-            self.entity_description.key,
-            self._attr_is_on,
-        )
+        self._attr_is_on = self.appliance.settings[SettingKey(self.bsh_key)].value
 
 
 class HomeConnectProgramSwitch(HomeConnectEntity, SwitchEntity):
     """Switch class for Home Connect."""
 
-    def __init__(self, device: HomeConnectDevice, program_name: str) -> None:
+    def __init__(
+        self,
+        coordinator: HomeConnectCoordinator,
+        appliance: HomeConnectApplianceData,
+        program: EnumerateAvailableProgram,
+    ) -> None:
         """Initialize the entity."""
-        desc = " ".join(["Program", program_name.split(".")[-1]])
-        if device.appliance.type == "WasherDryer":
+        desc = " ".join(["Program", program.key.split(".")[-1]])
+        if appliance.info.type == "WasherDryer":
             desc = " ".join(
-                ["Program", program_name.split(".")[-3], program_name.split(".")[-1]]
+                ["Program", program.key.split(".")[-3], program.key.split(".")[-1]]
             )
-        super().__init__(device, SwitchEntityDescription(key=program_name))
-        self._attr_name = f"{device.appliance.name} {desc}"
-        self._attr_unique_id = f"{device.appliance.haId}-{desc}"
+        super().__init__(
+            coordinator, appliance, SwitchEntityDescription(key=program.key)
+        )
+        self._attr_name = f"{appliance.info.name} {desc}"
+        self._attr_unique_id = f"{appliance.info.ha_id}-{desc}"
         self._attr_has_entity_name = False
-        self.program_name = program_name
+        self.program = program
 
-    async def async_added_to_hass(self) -> None:
+    async def async_added_to_hass(self) -> None:  # pylint: disable=hass-missing-super-call
```

Comment:
Why do we disable pylint here instead of solving the warning?

Reply:
Because the coordinator_context was based on the program key, so I didn't want the entity to be subscribed to a program key, it needed to be subscribed to a event key, specifically to the active program event key.
But I found another way to fix that, pushing it.